src/llvm-late-gc-lowering.cpp

```diff
@@ -727,8 +727,8 @@ State LateLowerGCFrame::LocalScan(Function &F) {
         for (auto it = BB.rbegin(); it != BB.rend(); ++it) {
             Instruction &I = *it;
             if (CallInst *CI = dyn_cast<CallInst>(&I)) {
-                if (isa<IntrinsicInst>(CI)) {
-                    // Intrinsics are never GC uses/defs
+                if (isa<IntrinsicInst>(CI) && !isa<MemIntrinsic>(CI)) {
```

Comment:
I assume this covers other intrinsics that touches memory? (like scatter, gather etc.)?

Reply:
And maybe add this condition back below where we check `pointer_from_objref` and `memcmp` so that they won't be treated as safepoints?

Reply:
Ah, I forgot about scatter and gather. There's probably also some machine specific intrinsic intrinsics that take memory operands. Probably easiest to just check debug and lifetime here (which are the only ones we emit frequently), but still declare all intrinsics as non-safepoint.